<a href="https://colab.research.google.com/github/Rizzamae09/1102_Test/blob/main/1102_Test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

MULTIPLE CHOICE

Which following is the correct way to create a dictionary in Python?

   (A) students = { 'name' => 'alice' ;  'score' =>60 }

   (B) students = { 'name' = 'alice' ;  'score' =60 }

   (C) students = { 'name' : 'alice' ;  'score' : 60 }

   (D) students = { 'name' => 'alice' ,  'score' =>60 }

   (E) students = { 'name' : 'alice' , 'score' : 60 }


In [10]:
{ 'name' : 'alice' , 'score' : 60 }

{'name': 'alice', 'score': 60}

Answer: (E)

What does the following Python program fragment do?

   s=0

   for i in range (3 , 10) :

        s+=i

   print (s)

   (A) Calculate the sum of 1+2+…+10

   (B) Calculate the sum of 1+4+7+10

   (C) Calculate the sum of 3+4+5+6+…+9

   (D) Calculate the sum of 3+4+5+6+…+10

   (E) Calculate the sum of 0+2+4+6+…+10
   

Answer: (C)

What is the output of this code in Python?

      list = [ ' a ' ,  ' b ' ,  ' c ' ,  ' d ' ,  ' e ' ]

      print (list [ 1 : ])

   (A) [ ]

   (B) [ 'b' ]

   (C) [ 'a' ,  'b' ]

   (D) [ 'b' ,  'c' ,  'd' ,  'e' ]
   

In [11]:
list = [ ' a ' ,  ' b ' ,  ' c ' ,  ' d ' ,  ' e ' ]
print(list [1:])

[' b ', ' c ', ' d ', ' e ']


Answer: (D)

What is the output of this code in Python?

        L1 = [ ]

        L1.append ( [1, [2, 3], 4] )

        L1.extend ( [7, 8, 9] )

        print(L1[0][1][1] + L1[2] )
        
   (A) 3  (B) 8  (C) 11  (D) 38  (E) None of these

In [12]:
L1 = [ ]

Answer: (E)

What is the output of this code in python?

        r = lambda q: q * 2

        s = lambda q: q * 3

        x = 2

        x = r (x)

        x = s (x)

        x = r (x)

        print (x)
        
   (A) 2  (B) 4  (C) 6  (D) 12  (E) 24

In [14]:
r = lambda q: q * 2
s = lambda q: q * 3
x = 2
x = r (x)
x = s (x)
x = r (x)
print (x)

24


ANswer: (E)

Which of the following conditions correctly describes the output of this code in  Python?

         first = [1 , 2 , 3 , 4 , 5]

         second = first

         second.append (6)

         print (first)

         print (second)

   (A) first=[1, 2, 3, 4, 5], second=[1, 2, 3, 4, 5, 6]

   (B) first=[1, 2, 3, 4, 5, 6], second=[1, 2, 3, 4, 5]

   (C) first=[1, 2, 3, 4, 5], second=[1, 2, 3, 4, 5]

   (D) first=[1, 2, 3, 4, 5, 6], second=[1, 2, 3, 4, 5, 6]
   
   (E) None of these

In [15]:
first = [1 , 2 , 3 , 4 , 5]
second = first
second.append (6)
print (first)
print (second)

[1, 2, 3, 4, 5, 6]
[1, 2, 3, 4, 5, 6]


Answer: (D)

If the function 

int volume (int x = 1, int y = 1, int z = 1); 
is called by the expression volume (7,8), 
how many default arguments are used?

(A) None  (B) One  (C) Two  (D) Three

(E) It depends on the runtime environment.

Answer :(A)

READING COMPREHENSION

What is the output of this code in Python?

# primes.py

primes = []  # this will contain the primes at the end

upto = 100  # the limit, inclusive

for n in range(2, upto + 1):

    is_prime = True  # flag, new at each iteration of outer for

    for divisor in range(2, n):

        if n % divisor == 0:

            is_prime = False

            break

    if is_prime:  # check on flag

        primes.append(n)
        
print(primes)

In [16]:
# primes.py
primes = []  # this will contain the primes at the end
upto = 100  # the limit, inclusive
for n in range(2, upto + 1):
    is_prime = True  # flag, new at each iteration of outer for
    for divisor in range(2, n):
        if n % divisor == 0:
            is_prime = False
            break
    if is_prime:  # check on flag
        primes.append(n)
print(primes)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


Answer: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]

What is the output of this code in Python?

In [17]:
# primes.else.py
primes = []
upto = 100
for n in range(2, upto + 1):
    for divisor in range(2, n):
        if n % divisor == 0:
            break
    else:
        primes.append(n)
print(primes)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


Answer: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]

What is the output of this code in Python?

In [18]:
cubes = [x**3 for x in range(1,6)]
print(cubes)

[1, 8, 27, 64, 125]


Answer: [1, 8, 27, 64, 125]

Programming Python

(1)Write a Python function/Program to Find Fibonacci Numbers using Recursion

In [21]:
def recur_fibo(n):  
   if n <= 1:  
       return n  
   else:  
       return(recur_fibo(n-1) + recur_fibo(n-2))  
# take input from the user  
nterms = int(input("How many terms? "))  
# check if the number of terms is valid  
if nterms <= 0:  
   print("Plese enter a positive integer")  
else:  
   print("Fibonacci numbers:")  
   for i in range(nterms):  
       print(recur_fibo(i))  

How many terms? 25
Fibonacci numbers:
0
1
1
2
3
5
8
13
21
34
55
89
144
233
377
610
987
1597
2584
4181
6765
10946
17711
28657
46368


(2)Use iterative Approach To Find the nth Term Of Fibonacci Sequence(You need to write a function using loop)

In [22]:
def fibonacci(n):
    a,b = 0,1
    for i in range(n):
        a,b = b,a+b
    return a

print(fibonacci(6))

8


In [ ]:
a = 0
b = 1
n=int(input("Enter the number of terms in the sequence: "))
print(a,b,end=" ")
while(n-2):
    c=a+b
    a,b = b,c
    print(c,end=" ")
    n=n-1